gateway-sp-comms/src/single_sp.rs

```diff
@@ -1048,6 +1137,168 @@ impl SingleSp {
+                None
+            }
+            Err(e) => return Err(e),
+        };
+
+        let key: u32 = rand::random();
+        let result = rpc(
+            &self.cmds_tx,
+            MgsRequest::Dump(DumpRequest::TaskDumpReadStart { key, index }),
+            None,
+        )
+        .await;
+
+        let task =
+            result.result.and_then(
+                |(_peer, response, data)| match response {
+                    SpResponse::Dump(DumpResponse::TaskDumpReadStarted(t)) => {
+                        if data.is_empty() {
+                            Ok(t)
+                        } else {
+                            Err(CommunicationError::UnexpectedTrailingData(
+                                data,
+                            ))
+                        }
+                    }
+                    SpResponse::Error(err) => {
+                        Err(CommunicationError::SpError(err))
+                    }
+                    other => Err(CommunicationError::BadResponseType {
+                        expected: "task_dump_read_start",
+                        got: other.into(),
+                    }),
+                },
+            )?;
+        debug!(self.log, "got task {task:?}");
+
+        let mut map: BTreeMap<u32, Vec<u8>> = BTreeMap::new();
+        loop {
+            let result = rpc(
+                &self.cmds_tx,
+                MgsRequest::Dump(DumpRequest::TaskDumpReadContinue { key }),
+                None,
+            )
+            .await;
+
+            let r = result.result.and_then(
+                |(_, response, data)| match response {
+                    SpResponse::Dump(DumpResponse::TaskDumpRead(None)) => {
+                        Ok(None)
+                    }
+                    SpResponse::Dump(DumpResponse::TaskDumpRead(Some(s))) => {
+                        if data.len() != s.compressed_length as usize {
+                            Err(CommunicationError::BadTrailingDataSize {
+                                expected: s.compressed_length as usize,
+                                got: data.len(),
+                            })
+                        } else {
+                            Ok(Some((s, data)))
+                        }
+                    }
+                    SpResponse::Error(err) => {
+                        Err(CommunicationError::SpError(err))
+                    }
+                    other => Err(CommunicationError::BadResponseType {
+                        expected: "task_dump_read",
+                        got: other.into(),
+                    }),
+                },
+            )?;
+            if let Some((header, data)) = r {
+                debug!(
+                    self.log,
+                    "got {} bytes from {:#08x}",
+                    data.len(),
+                    header.address
+                );
+                // The decompressor type must agree with `humpty`
```

Comment:
Should one of the responses (probably to dump start?) have a field for what kind of compression was used, even if that field currently only has one possible value?

Reply:
Good idea, added in 6cf8e7c1